docs/modules/traits/pages/pod.adoc

```diff
@@ -0,0 +1,83 @@
+= Pod Trait
+
+// Start of autogenerated code - DO NOT EDIT! (description)
+   The pod trait allows the customization of the Integration pods.
+   It applies the `PodSpecTemplate` struct contained in the Integration `.spec.podTemplate` field, into the Integration deployment Pods template, using strategic merge patch.
+
+   This can be used to customize the container where Camel routes execute, by using the `integration` container name.
+
+
+This trait is available in the following profiles: **Kubernetes, Knative, OpenShift**.
+
+// End of autogenerated code - DO NOT EDIT! (description)
+== Example
+We have an integration that will read files from defined folder:
+[source,groovy]
+----
+from('file:///var/log')
+ .convertBodyTo(String.class)
+ .setBody().simple('${body}:  {{TEST_VARIABLE}} ')
+ .log('${body}')
+
+----
+The content of the folder might be autogenerated by bash script using a sidecar container. The pod template could be defined as follows:
+
+[source,yaml]
+----
+containers:
+  - name: integration
```

Comment:
How would this work when you have competing changes to the integration container ?
i.e. 

1. what would be the result of `kamel run --env TEST_VARIABLE=abc --pod-template template.yaml` ?
2. what woudl happen if the user sets a custom entry point for the integration container ?

Reply:
1. TEST_VARIABLE=abc will be overwritten by values from the template, due to changes from the template are actually applied by pod trait in later stages. 
2. I've tried to add there command from the example: `command: [ "/bin/sh" , "-c", "while true; do echo $(date -u) 'Content from the sidecar altered container' > /var/log/file.txt; sleep 1;done" ]` and nothing really happened. Command hasn't been overwritten.

Reply:
So I think the behavior is ok for the sake of this PR to get merged but I think we need to open some follow up issues and iterate a little bit over the expected behavior as:
- the pod-template is well, a template so I'd expect it not to be "immutable" so, in the cli example I provided above, I'd expect the `TEST_VARIABLE` to reflect what it is set by the `--env` flag and not the value from the template. Of course in this case, you know what the template content is so you could workaround this but I'm thinking about a case where the template is configured at an higher level (i.e. IntegrationPlatform) and you don't know if you are overwriting, or if your changes have no effect because the template is overriding some of your config
- the fact that a `TEST_VARIABLE` is taken from the template and the `command` not, make things a little confusing so we should probably limit/document the relation between the template and the `integration` container

Reply:
Right, we need to decide what is the precedence. I see three layers:
* operator
* user provided pod template
* any other user provided options (that affects the integration pod)

The first decision to make is whether the operator is prescriptive or not, that is whether it's possible to override the operator defined spec by the template. Then whether the pod template overwrites other user defined options, or the other way around. For the later, I would be inclined to think it's better from an end-user standpoint to have the specific options have precedence over the more general pod template. For the former, that depends on the level of power we want to delegate to the users.

Reply:
So for now I will add notes into the doc about env variables and the command option.  And if this will get merged we might create a follow-up issue to address env variables and other options that could be possibly overwritten by the template option.